src/App.tsx

```diff
@@ -1,61 +1,138 @@
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  const [showTitleError, setShowTitleError] = useState(false);
+  const [showUserError, setShowUserError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const validTitle = event.target.value.replace(/[^a-zA-Za-яA-Я ]/g, '');
+
+    setNewTodoTitle(validTitle);
+  };
+
+  const addNewTodo = (title: string, userId: number, id: number) => {
+    const newTodo: Todo = {
+      id: id + 1,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const currentMaxId = Math.max(...todos.map(todo => todo.id));
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setShowTitleError(!newTodoTitle);
+    setShowUserError(!selectedUserId);
+
+    if (newTodoTitle && selectedUserId) {
+      addNewTodo(newTodoTitle, selectedUserId, currentMaxId);
+    }
+  };
+
+  useEffect(() => {
+    setNewTodoTitle('');
+    setSelectedUserId(0);
+  }, [todos.length]);
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <header className="header">
+        <div className="header-content">
+          <h1 className="header__title">
+            Add todo form
+          </h1>
 
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
+          <form
+            action="/api/users"
+            method="POST"
+            className="header__form"
+            onSubmit={handleFormSubmit}
+          >
+            <div className="form-content">
+              <div className="field form-field">
+                <input
+                  type="text"
+                  className="header__field input"
+                  data-cy="titleInput"
+                  placeholder="Enter a title"
+                  value={newTodoTitle}
+                  onChange={handleTitleChange}
+                />
+
+                {showTitleError
+                  && !newTodoTitle
+                    && (
+                      <div>
+                        <span className="error">{Error.TitleAbsence}</span>
+                      </div>
+                    )}
+              </div>
+
+              <div className="field">
+                <select
+                  data-cy="userSelect"
+                  className="header__field select"
+                  value={selectedUserId}
+                  onChange={event => (
+                    setSelectedUserId(Number(event.target.value)))}
+                >
+                  <option value="0" disabled>Choose a user</option>
+                  {usersFromServer.map((user) => (
+                    <option key={user.id} value={user.id}>
```

Comment:
Consider using a more descriptive variable name instead of `user`. For example, `selectedUser`. 🤔